### ISSUE TITLE
Copy of Code quality review - Proofing Agent Facing Screens

### ISSUE THREAD

**Issue by SophiaPhilipMO** [department-of-veterans-affairs/va.gov-team#79104]
Who: Frontend engineer

When: As part of daily development

Semantic HTML is the cornerstone of accessibility, and ARIA attributes must be used with care. Periodic code quality reviews by frontend engineers can prevent incompatibilities with assistive technologies.

#### [](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#Prepareforanaccessibilitystagingreview-Codequalitychecks "Copy to clipboard")Code quality checks

-   Confirm that any Design System components are used as intended, according to Design System documentation.

-   Confirm that all HTML elements are implemented consistent with specifications and best practices, including:

    -   Element IDs are unique.

    -   Links are navigational and are coded using the `a` element.

    -   Buttons are actions and are coded using the `button` element.

    -   Form input are properly labeled and, when appropriate, grouped in a `fieldset` with related inputs.

    -   [Visible input labels](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html) match the `name` attribute.

    -   Form inputs have the appropriate input `type` attribute so that the [purpose of the input can be determined programmatically (when applicable)](https://www.w3.org/TR/WCAG21/#x7-input-purposes-for-user-interface-components).

    -   [Status messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) have the appropriate `role` attribute.

-   Confirm that all ARIA attributes are implemented according to the W3C's [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/).

#### [](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#Prepareforanaccessibilitystagingreview-Definitionofdone.6 "Copy to clipboard")Definition of done

Design System components, HTML elements, and ARIA attributes are all implemented consistent with documentation and best practices.

If there are code quality issues that you are unable to resolve prior to your staging review, make sure these are documented in your accessibility testing artifact.

### [](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#Prepareforanaccessibilitystagingreview-Mouse-onlyandtouchscreenmouse-and-touch "Copy to clipboard")
